dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterService.java

```diff
@@ -201,10 +202,10 @@ private Event from(
       result.setUid(!StringUtils.isEmpty(event.getEvent()) ? event.getEvent() : event.getUid());
       result.setCreated(now);
       result.setStoredBy(event.getStoredBy());
-      result.setCreatedByUserInfo(preheat.getUserInfo());
+      result.setCreatedByUserInfo(UserInfoSnapshot.from(CurrentUserUtil.getCurrentUserDetails()));
```

Comment:
I am not sure how some of these classes are currently tested. Changes like this make them less easy to unit test. Not sure what the strategy is but if we want to unit test such transformations we should pass the user into the function.

Reply:
Maybe we can discuss this and agree on the best way to use the user and have testable code.
At the moment, we need to inject the user in the tests.